### DISCUSSION TITLE
When time::serde::format_description defined submodule::option returns None?

### DISCUSSION THREAD

**Discussion by Neutroni** [time-rs/time#460] [Question] [answered]
I am trying to parse some timestamps that might not be present and trying to use `#[serde(with = "my_format::option")]` to indicate that the value might not be present but it seems that `Option<Date>` behaves differently than at least `Option<String>` when de-serializing, this might not necessarily be a bug but I want to know under what conditions the de-serialized `Option<Date>` can be None.
Following is my code that I expected to run but panics because the field is missing.
```
use time::PrimitiveDateTime;
use serde::{Deserialize, Serialize};

// Makes a module `mod my_format { ... }`.
time::serde::format_description!(my_format, PrimitiveDateTime, "[year]-[month]-[day] [hour]:[minute]:[second].[subsecond]");

#[derive(Serialize, Deserialize, Debug)]
struct SerializesWithCustom {
    #[serde(with = "my_format")]
    dt: PrimitiveDateTime,
    #[serde(with = "my_format::option")]
    maybe_dt: Option<PrimitiveDateTime>,
}

fn main() {
    let data = r#"
        {
            "dt": "2022-01-01 12:00:00.0"
        }"#;

    // Parse the string of data into serde_json::Value.
    let v: SerializesWithCustom = serde_json::from_str(data)
        .expect("Failed to parse input");
    println!("{:?}", v);
}
```
Chancing the `Option<PrimitiveDateTime>` to `Option<String>` and removing the `#[serde(with = "my_format::option")]` causes
the code to behave as I would expect.

Dependencies:
```
serde = { version = "1.0.136", features = ["derive"] }
serde_json = "1.0.79"
time = { version = "0.3.9", features = ["macros", "serde-human-readable", "serde"] }
```

In case anyone finds this and wonders how to fix the issue add the attribute `default` to the field, keeping the question open since the behaviour without the attribute is unexpected, might be documentation issue assuming not a bug.

**Comment by jhpratt** [comment, marked as answer]
Deserializing `Option<Date>` will be `None` if the value is `null` (in JSON — other formats have their own equivalents). When this was brought up before, I was unable to find a way to fall back to `None` if the value is not present. You'd have to use `#[serde(default)]` on the field as well. The name may be different, as I'm typing this from memory.

**Reply by Neutroni** [reply]
Thank you, Originally I was working with TOML which as far as I know does not have null. It is possible that
https://docs.rs/time/0.3.9/time/serde/macro.format_description.html
needs to be updated to document this behaviour. Maybe something like:
The returned Option will contain de-serialized value if present, None if field is present and contains a "null" value, to de-serialize missing fields as None add `#[serde(default)]` annotation.
